buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java

```diff
@@ -0,0 +1,218 @@
+            });
+        }
+        // only include x-pack if explicitly instructed
+        if (includeXpack.get().isEmpty() == false) {
+            logger.info("X-pack YAML{} for project [{}] will be copied to the test resources.", getTestOrSpec(), project.getPath());
+            project.copy(c -> {
+                c.from(xpackConfig.getSingleFile());
+                c.into(getOutputDir());
+                c.include(xpackPatternSet.getIncludes());
+            });
+        }
+    }
+
+    /**
+     * Returns true if any files with a .yml extension exist the test resources rest-api-spec/test directory (from source or output dir)
+     */
+    private boolean projectHasYamlRestTests() {
+        File testSourceResourceDir = getTestSourceResourceDir();
+        File testOutputResourceDir = getTestOutputResourceDir(); // check output for cases where tests are copied programmatically
+
+        if (testSourceResourceDir == null && testOutputResourceDir == null) {
+            return false;
+        }
+        try {
+            if (testSourceResourceDir != null) {
+                return new File(testSourceResourceDir, "rest-api-spec/test").exists() == false
+                    || Files.walk(testSourceResourceDir.toPath().resolve("rest-api-spec/test"))
+                        .anyMatch(p -> p.getFileName().toString().endsWith("yml"));
+            }
+            if (testOutputResourceDir != null) {
+                return new File(testOutputResourceDir, "rest-api-spec/test").exists() == false
+                    || Files.walk(testOutputResourceDir.toPath().resolve("rest-api-spec/test"))
+                        .anyMatch(p -> p.getFileName().toString().endsWith("yml"));
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException(String.format("Error determining if this project [%s] has rest tests.", getProject()), e);
+        }
+        return false;
+    }
+
+    private File getTestSourceResourceDir() {
+        SourceSet testSources = getTestSourceSet();
+        if (testSources == null) {
+            return null;
+        }
+        Set<File> resourceDir = testSources.getResources()
+            .getSrcDirs()
+            .stream()
+            .filter(f -> f.isDirectory() && f.getParentFile().getName().equals("test") && f.getName().equals("resources"))
+            .collect(Collectors.toSet());
+        assert resourceDir.size() <= 1;
+        if (resourceDir.size() == 0) {
+            return null;
+        }
+        return resourceDir.iterator().next();
+    }
+
+    private File getTestOutputResourceDir() {
+        SourceSet testSources = getTestSourceSet();
+        if (testSources == null) {
+            return null;
+        }
+        return testSources.getOutput().getResourcesDir();
+    }
+
+    private SourceSet getTestSourceSet() {
+        return Boilerplate.getJavaSourceSets(getProject()).findByName("test");
+    }
+
+    private String getTestOrSpec() {
```

Comment:
Let's not trigger behavior based on loose conventions. Let's make this an explicit thing.